tests/test_validation_pandapipes.py

```diff
@@ -0,0 +1,268 @@
+        ds_demand_power_watt = DFData(demand_power_watt)
+        control.ConstControl(
+            net,
+            element="heat_exchanger",
+            variable="qext_w",
+            element_index=net.heat_exchanger.index.values[idemand],
+            data_source=ds_demand_power_watt,
+            profile_name=net.heat_exchanger.index.values[idemand].astype(str),
+        )
+
+        demand_flow_kgs_s = pd.DataFrame(
+            demand_flow,
+            columns=[f"{idemand + total_producers}"],
+        )
+        ds_demand_control_flow_kgs_s = DFData(demand_flow_kgs_s)
+        control.ConstControl(
+            net,
+            element="flow_control",
+            variable="controlled_mdot_kg_per_s",
+            element_index=net.flow_control.index.values[idemand + total_producers],
+            data_source=ds_demand_control_flow_kgs_s,
+            profile_name=net.flow_control.index.values[idemand + total_producers].astype(str),
+        )
+
+        if profile_demand_load_watt.shape[0] != supply_flow_kg_s.shape[0]:
+            exit("profiles do not match")
+        # Completed scenario setup
+
+        # ------------------------------------------------------------------------------
+        # ------------------------------------------------------------------------------
+        # Run panda_pipes simulation
+        time_steps = range(profile_demand_load_watt.shape[0])
+        log_variables = [
+            ("res_pipe", "v_mean_m_per_s"),
+            ("res_pipe", "p_from_bar"),
+            ("res_pipe", "p_to_bar"),
+            ("res_pipe", "mdot_from_kg_per_s"),
+            ("heat_exchanger", "qext_w"),
+        ]
+        ow = OutputWriter(
+            net,
+            time_steps,
+            output_path=None,
+            log_variables=log_variables,
+        )
+
+        try:
+            run_timeseries(net, time_steps, mode="all", friction_model="colebrook")
+        except Exception as e:
+            print(e)
+            exit("Pandapipes runs was not successful")
+
+        # ------------------------------------------------------------------------------
+        # Post processing panda_pipes results and tests
+        net = esdlparser.correcting_pressure_return(net)
+        ow.np_results["heat_exchanger.qext_w"]
+        mdata_points = len(mesido_demand_flow_kg_s)
+
+        density = (
+            ow.np_results["res_pipe.mdot_from_kg_per_s"][0:mdata_points]
+            / ow.np_results["res_pipe.v_mean_m_per_s"][0:mdata_points]
+            / (np.pi * net.pipe.diameter_m[0] * net.pipe.diameter_m[0] / 4.0)
+        )
+        pandapipes_head_loss_m = (
+            (
+                ow.np_results["res_pipe.p_to_bar"][0:mdata_points]
+                - ow.np_results["res_pipe.p_from_bar"][0:mdata_points]
+            )
+            * 100.0e3
+            / density[0][0]
+            / GRAVITATIONAL_CONSTANT
+        )
+
+        # Compare head losses
```

Comment:
I would suggest that we include the plotting code (commented), so that people can locally run validation and visualize easily

Reply:
thnx looks good